### PR TITLE
Partition mode

### PR DESCRIPTION
**Pull request by bcl**
This is basically a port of https://github.com/osbuild/osbuild-composer/pull/3222 to images. I *think* this is complete, but I need to clean up the osbuild-composer part first so marking this as a draft for now.

### CODE REVIEW

**Comment by bcl** [comment]
This is ready to review. It is currently vendored into https://github.com/osbuild/osbuild-composer/pull/3723 to see if the full suite of tests will pass with it.

It's slightly different from the original PR, I decided to make an empty/missing PartitioningMode use DefaultPartitioningMode so that the string conversions are nicer to read. The fact that Default results in AutoLVM is enforced in this code and nowhere else. eg. If the osbuild-composer code gets an empty entry it calls this code with DefaultPartitioningMode instead of AutoLVM so that we have a single place deciding what the default actually is.

**Comment by ondrejbudai** [comment]
Awesome, @bcl! I will wait for Achilleas to return, I don't feel comfortable giving this an approval, considering that I wrote most of these changes. :)

File: pkg/distro/fedora/imagetype.go [hunk 1]
```diff
@@ -136,9 +136,18 @@ func (t *imageType) getPartitionTable(
...
-	lvmify := !t.rpmOstree
+	partitioningMode := options.PartitioningMode
+	if t.rpmOstree {
+		// IoT supports only LVM, force it.
+		// LVM is not supported, return an error if it is requested
+		// TODO Need a central location for logic like this
+		if partitioningMode == disk.RawPartitioningMode {
+			return nil, fmt.Errorf("partitioning mode raw not supported for %s on %s", t.Name(), t.arch.Name())
+		}
+		partitioningMode = disk.AutoLVMPartitioningMode
```
[the remark this answers is not in the document]
Comment (bcl) [reply pkg/distro/fedora/imagetype.go:147]: oops, I didn't update the comment correctly. The behavior changed between the time of the first version of this from @ondrejbudai and now -- IoT now always uses LVM, it used to be the other way around (and the tests caught that but I didn't clean up all the comments).

**Review by achilleas-k — APPROVED** [review]

**Comment by achilleas-k** [comment]
THANKS! <3

**Pull request merged**